packages/build-tools/src/customBuildContext.ts

```diff
@@ -76,4 +76,14 @@ export class CustomBuildContext implements ExternalBuildContextProvider {
   public updateEnv(env: Env): void {
     this._env = env;
   }
+
+  public updateJobInformation(job: Job, metadata: Metadata): void {
+    if (this.job.triggeredBy !== BuildTrigger.GIT_BASED_INTEGRATION) {
+      throw new Error(
+        'Updating job information is only allowed when build was triggered by a git-based integration.'
+      );
+    }
+    this.job = { ...job, triggeredBy: this.job.triggeredBy };
```

Comment:
Is this the only property we may lose information about?

Reply:
Yeah, I think so